### PR TITLE
Dev/woocommerce subscriptions

### PR DESCRIPTION
**Pull request by MichaelM5**

### CODE REVIEW

File: src/trait-wc-gateway-cp-subscriptions.php [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            $accountNo = get_post_meta($subscription_id, 'AccountNo', true);
+            $account = $this->account_retrieval($accountNo);
+
+            $token = array_values($account['CardHolderAccount']['cards'])[0]['token'];
+
+            $charge_body = [
+                "amount" => $amount_to_charge,
+                "identifier" => "renewal-" . $merchant_id . $subscription_id . $renewal_order_id,
+                "merchantid" => (int) $merchant_id ,
+                "token" => $token,
```
Comment (garyfeltham) [inline src/trait-wc-gateway-cp-subscriptions.php:75]: should this not include subscription_id in the charge body?

File: src/trait-wc-citypay-api.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+        $context = get_file_data(__DIR__ . '/wc-payment-gateway-citypay.php', ['version' => 'Version']);
+
+        $user_agent = 'WooCommerce-' . wc()->version . '/CityPay-WC-' . $context['version'];
+
+        $body = [
+            "amount" => $charge['amount'],
+            "identifier" => $charge['identifier'],
+            "merchantid" => $charge['merchantid'],
+            "token" => $charge['token'],
```
Comment (garyfeltham) [inline src/trait-wc-citypay-api.php:84]: should this include the subscription_id value?

File: src/trait-wc-gateway-cp-subscriptions.php (hunk 1 above)
Comment (hectorconceicao) [reply src/trait-wc-gateway-cp-subscriptions.php:75]: It should. I've just added it.

File: src/trait-wc-citypay-api.php (hunk 2 above)
Comment (hectorconceicao) [reply src/trait-wc-citypay-api.php:84]: It should. I've just added it.

**Review by garyfeltham — APPROVED** [review]

**Pull request merged**